### PR TITLE
Missing validators on one level should go to the next validators set

### PR DESCRIPTION
**Pull request by andralungu**

### CODE REVIEW

File: src/lua/api-gateway/validation/validatorsHandler.lua [hunk 1]
```diff
@@ -185,7 +185,6 @@ function ValidatorsHandler:validateSubrequests(order, subrequests, localContext,
 
     if (subrequests_count == 0) then
         ngx.log(ngx.WARN, "Nothing to validate on subrequest")
-        ngx.exit(ngx.HTTP_OK)
```
Comment (constantincristian) [inline src/lua/api-gateway/validation/validatorsHandler.lua]: Can we add a unit/integration test that would demonstrate this indeed fixes the issue?

**Review by constantincristian — CHANGES_REQUESTED** [review]

**Comment by andralungu** [comment]
@constantincristian, I was trying to solve the task one subtask at a time 😄 
Please have another look.

File: src/lua/api-gateway/validation/validatorsHandler.lua [hunk 2]
```diff
@@ -243,7 +233,7 @@ function ValidatorsHandler:saveContextInRequestVars(localContext)
     -- for i,k in pairs(varsToSet) do
         if ngx.var[k] ~= nil and (type(localContext[k]) == "string" or type(localContext[k]) == "number") then
             -- ngx.log(ngx.DEBUG, "Setting " .. k .. ",from: " .. ngx.var[k] .. ",to:" .. v)
-            ngx.var[k] = localContext[k]
+            ngx.var[k] = v
```
Comment (atrifan) [inline src/lua/api-gateway/validation/validatorsHandler.lua]: 👍

File: src/lua/api-gateway/validation/validatorsHandler.lua [hunk 3]
```diff
@@ -254,19 +244,22 @@ function ValidatorsHandler:validateRequest()
     local subrequestResultStatus = ngx.HTTP_OK
     local subrequestResultBody
     local responseTimesHeaders = {}
-    local reqs_count = table.getn(reqs)
+    local reqs_count = table.maxn(reqs)
```
Comment (atrifan) [inline src/lua/api-gateway/validation/validatorsHandler.lua:249]: 👍

**Review by atrifan — APPROVED** [review]

**Review by vdatcu — APPROVED** [review]

File: src/lua/api-gateway/validation/validatorsHandler.lua (hunk 2 above)
Comment (atrifan) [inline src/lua/api-gateway/validation/validatorsHandler.lua]:
```
if v ~= ngx.null then
                ngx.var[k] = v
            end
```

**Review by atrifan — CHANGES_REQUESTED** [review]

**Comment by andralungu** [comment]
@atrifan please have another look

**Pull request merged**